docs/backend-system/core-services/01-index.md

```diff
@@ -65,10 +88,366 @@ createBackendPlugin({
+
+createBackendPlugin({
+  id: 'example',
+  register(env) {
+    env.registerInit({
+      deps: {
+        lifecycle: coreServices.lifecycle,
+        logger: coreServices.logger,
+      },
+      async init({ lifecycle, logger }) {
+        // setup by creating an interval that does something that we want to stop after the plugin is stopped.
+        const interval = setInterval(async () => {
+          await fetch('http://google.com/keepalive').then(r => r.json());
+          // do some other stuff.
+        });
+
+        lifecycle.addShutdownHook({
+          fn: () => clearInterval(interval),
+          logger,
+        });
+      },
+    });
+  },
+});
+```
+
+## Permissions
+
+Sometimes you want to include permissions and making sure that a user that is authorized to do some actions in your plugin. We've provide a core service out of the box for you to interact with the permissions framework. You can find out more about the permissions framework in [the documentation](https://backstage.io/docs/permissions/overview)
+
+### Using the service
+
+The following example shows how to get the `PermissionsSerice` in your `example` backend to check to see if the user has the correct permissions for `myCustomPermission`.
+
+```ts
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { Router } from 'express';
+
+createBackendPlugin({
+  id: 'example',
+  register(env) {
+    env.registerInit({
+      deps: {
+        permissions: coreServices.permissions,
+        http: coreServices.httpRouter,
+      },
+      async init({ permissions, http }) {
+        const router = Router();
+        router.get('/test-me', (request, response) => {
+          // use the identityService pull out the header from the request and get the token
+          const { token } = await identity.getIdentity({
+            request,
+          });
+
+          // ask the permissions framework what the decision is for the permission
+          const permissionResponse = await permissions.authorize(
+            [
+              {
+                permission: myCustomPermission,
+              },
+            ],
+            { token },
+          );
+        });
+
+        http.use(router);
+      },
+    });
+  },
+});
+```
+
+## Scheduler
+
+When writing plugins, it's often that you want to have things running on a schedule, or something similar to cron jobs that are distributed through instances that your backend plugin might be running on. We supply a `TaskScheduler` that is scoped per plugin so that you can create these tasks and orchestrate the running of them.
+
+### Using the service
+
+The following example shows how to get the `SchedulerService` in your `example` backend to schedule a scheduled task that runs once across your instances at a given interval.
+
+```ts
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { fetch } from 'node-fetch';
```

Comment:
remove

Reply:
Answered in https://github.com/backstage/backstage/pull/15851#discussion_r1082442134